### PR TITLE
Feature/output handler pretty

### PR DESCRIPTION
**Pull request by rinslow**
Added an (on-by-default) output handler called "pretty". 
It decorates the beginning and ending of tests, creating an easy-to-identify patterns in the log.

### CODE REVIEW

**Comment by osherdp** [comment]
Bump minor

File: src/rotest/core/result/handlers/stream/log_handler.py [hunk 1]
```diff
@@ -45,3 +46,29 @@ class LogDebugHandler(LogStreamHandler):
...
+
+
+class PrettyHandler(LogStreamHandler):
+    NAME = 'pretty'
+    LEVEL = logging.DEBUG
+
+    def start_test(self, test):
+        self.stream.writeln(str(Pretty(test, TestResult.started)))
+
+    def stop_test(self, test):
```
[the remark this answers is not in the document]
Comment (rinslow) [reply src/rotest/core/result/handlers/stream/log_handler.py]: k

File: src/rotest/core/result/handlers/stream/log_handler.py [hunk 2]
```diff
@@ -45,3 +46,29 @@ class LogDebugHandler(LogStreamHandler):
     """Debug-level log to screen handler."""
     NAME = 'logdebug'
     LEVEL = logging.DEBUG
+
+
+class PrettyHandler(LogStreamHandler):
```
[the remark this answers is not in the document]
Comment (rinslow) [reply src/rotest/core/result/handlers/stream/log_handler.py]: k

File: src/rotest/core/utils/pretty.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+            result=self.test_result(),
+            decoration_right=self.right_decoration()
+        )
+
+    def left_decoration(self):
+        decoration_color = self.configuration.decoration_color()
+        decoration_char = self.configuration.decoration_character()
+        left_decoration_length = self._whole_decoration_line_width() / 2
+
+        return colored(left_decoration_length * decoration_char,
```
[the remark this answers is not in the document]
Comment (rinslow) [reply src/rotest/core/utils/pretty.py]: k

File: src/rotest/core/utils/pretty.py [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+    if sys.platform in ("linux", "linux2", "darwin"):
+        # Both OS X (darwin) and Linux support the 'stty' command
+        _, columns = os.popen("stty size").read().split()
+        columns = int(columns)
+        return columns
+
+    if sys.platform == "win32":
+        from ctypes import windll, create_string_buffer
+        handle = windll.kernel32.GetStdHandle(-12)  # stderr
+        info_buffer = create_string_buffer(22)
```
[the remark this answers is not in the document]
Comment (rinslow) [reply src/rotest/core/utils/pretty.py]: 22 is just the size in bytes of the ConsoleScreenBufferInfo object

File: src/rotest/core/utils/pretty.py [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+        result = windll.kernel32.GetConsoleScreenBufferInfo(handle,
+                                                            info_buffer)
+        if result:
+            parsed_result = struct.unpack("hhhhHhhhhhh", info_buffer.raw)
```
[the remark this answers is not in the document]
Comment (rinslow) [reply src/rotest/core/utils/pretty.py]: I looked through and through for ways to get console size in win32, I couldn't find anything that worked. I'll push now but I'll try again

File: src/rotest/core/utils/pretty.py [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+    def test_name_color(self):
+        """Define the color of the test's name.
+
+        Returns:
+            str. name of the color to color the test name.
+        """
+        pass
+
+    @abstractmethod
+    def has_multi_line_decoration(self):
```
[the remark this answers is not in the document]
Comment (rinslow) [reply src/rotest/core/utils/pretty.py]: k

File: src/rotest/core/utils/pretty.py [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+
+class NonComplexTitleConfiguration(TitleConfiguration):
+    def has_multi_line_decoration(self):
+        return False
+
+    def decoration_character(self):
+        return "-"
+
+    def decoration_color(self):
+        colors = {
```
[the remark this answers is not in the document]
Comment (rinslow) [reply src/rotest/core/utils/pretty.py]: They're duplicated for the sake of being easy to configure the way pretty looks like, 
`namedtuple` solution would be shorter but I think harder to edit and understand what's going on.

File: src/rotest/core/utils/pretty.py [hunk 8]
```diff
@@ -0,0 +1,393 @@
...
+            return ComplexTitleConfiguration(result)
+
+        return NonComplexTitleConfiguration(result)
+
+
+class NonComplexTitleConfiguration(TitleConfiguration):
+    """Title configuration for tests that are not complex."""
+
+    @property
+    def has_multi_line_decoration(self):
```
Comment (gregoil) [inline src/rotest/core/utils/pretty.py]: ```has_multi_line_decoration = False```

File: src/rotest/core/utils/pretty.py [hunk 9]
```diff
@@ -0,0 +1,393 @@
...
+
+    def __eq__(self, other):
+        return str(self) == other
+
+
+class Pretty(object):
+    """Title with decoration.
+
+    Attributes:
+        test (AbstractTest): encapsulated test to generate it's title.
```
Comment (gregoil) [inline src/rotest/core/utils/pretty.py]: *its

File: src/rotest/core/utils/pretty.py [hunk 10]
```diff
@@ -0,0 +1,358 @@
...
+
+    def _test_name_uncolored(self):
+        return self.test.__class__.__name__
+
+    def _test_result_uncolored(self):
+        return str(self.result).capitalize()
+
+    def _whole_decoration_line_width(self):
+        """Return the width of the whole decoration line.
+            That means the left and right decoration combined.
```
[the remark this answers is not in the document]
Comment (gregoil) [reply src/rotest/core/utils/pretty.py]: Try again

File: src/rotest/core/utils/pretty.py (hunk 8 above)
Comment (rinslow) [reply src/rotest/core/utils/pretty.py]: ok

File: src/rotest/core/utils/pretty.py (hunk 9 above)
Comment (rinslow) [reply src/rotest/core/utils/pretty.py]: lol k

File: src/rotest/core/utils/pretty.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (rinslow) [reply src/rotest/core/utils/pretty.py]: I assume you want 4 characters from the first letter then... alright

File: src/rotest/core/utils/pretty.py [hunk 11]
```diff
@@ -0,0 +1,379 @@
...
+        if test.IS_COMPLEX:
+            return ComplexTitleConfiguration(result)
+
+        return NonComplexTitleConfiguration(result)
+
+
+class NonComplexTitleConfiguration(TitleConfiguration):
+    """Title configuration for tests that are not complex."""
+    @property
+    def decoration_character(self):
```
Comment (osherdp) [inline src/rotest/core/utils/pretty.py]: decoration_character = "-"

File: src/rotest/core/utils/pretty.py [hunk 12]
```diff
@@ -0,0 +1,379 @@
...
+
+    def _test_name_uncolored(self):
+        return self.test.__class__.__name__
+
+    def _test_result_uncolored(self):
+        return str(self.result).capitalize()
+
+    def _whole_decoration_line_width(self):
+        """Return the width of the whole decoration line.
+               That means the left and right decoration combined.
```
Comment (osherdp) [inline src/rotest/core/utils/pretty.py]: Deindent from this point, in the docstring

**Pull request merged**